packages/react-next/src/components/Slider/Slider.tsx

```diff
@@ -1,14 +1,74 @@
 import * as React from 'react';
-import { styled } from '../../Utilities';
-
+import { styled, css, memoizeFunction } from '../../Utilities';
 import { ISliderProps, ISliderStyleProps, ISliderStyles } from './Slider.types';
-
 import { SliderBase } from './Slider.base';
-import { getStyles } from './Slider.styles';
+import { getGlobalClassNames, ITheme } from '../../Styling';
+import * as classes from './Slider.scss';
+
+const GlobalClassNames = {
+  root: 'ms-Slider',
+  enabled: 'ms-Slider-enabled',
+  disabled: 'ms-Slider-disabled',
+  row: 'ms-Slider-row',
+  column: 'ms-Slider-column',
+  container: 'ms-Slider-container',
+  slideBox: 'ms-Slider-slideBox',
+  line: 'ms-Slider-line',
+  thumb: 'ms-Slider-thumb',
+  activeSection: 'ms-Slider-active',
+  inactiveSection: 'ms-Slider-inactive',
+  valueLabel: 'ms-Slider-value',
+  showValue: 'ms-Slider-showValue',
+  showTransitions: 'ms-Slider-showTransitions',
+  zeroTick: 'ms-Slider-zeroTick',
+};
+
+const getStaticStylesMemoized = memoizeFunction(
+  (
+    theme: ITheme,
+    className: string | undefined,
+    disabled: boolean | undefined,
+    vertical: boolean | undefined,
+    titleLabelClassName: string | undefined,
+    showTransition: boolean | undefined,
+    showValue: boolean | undefined,
+  ) => {
+    const globalClassNames = getGlobalClassNames(GlobalClassNames, theme);
+
+    const rootPropClasses = [
+      disabled ? classes.disabled : classes.enabled,
```

Comment:
I don't think the concept of `enabled` exists in the original implementation.

Reply:
It's there as a global class name and isn't one of the internal ones. I prefer the explicit use of checking for enabled vs the :not selector but that's really just my style. @xugao Do you have  an opinion of using the :not selector vs using an extra className

Reply:
let's have one prop -> one modifier class mapping. for now at least - this can be subject to change. i think the more modifier classes we have, the more complex things can get in the css land. so i'd like to have the fewer the better. another reason would we we can do something generic for all components when creating these prop -> class mapping when key is the same:
```
    const propControlledClasses = [
      disabled && classes.disabled,
      reversed && classes.reversed,
      checked && classes.checked,
      indeterminate && classes.indeterminate,
      isUsingCustomLabelRender && classes.isUsingCustomLabelRender,
    ];
```
I get that using `:not` can feel little funky, but there shouldn't be a strong downside for that. it doesn't increase specificity and perf wise should make no difference.